packages/grid/x-grid/src/tests/sorting.XGrid.test.tsx

```diff
@@ -216,56 +215,52 @@ describe('<XGrid /> - Sorting', () => {
 
       const t0 = performance.now();
       fireEvent.click(header);
-      await waitFor(() =>
-        expect(document.querySelector('.MuiDataGrid-sortIcon')).to.not.equal(null),
-      );
+      await waitFor(() => expect(document.querySelector('.MuiDataGrid-sortIcon')).to.not.be.null);
       const t1 = performance.now();
       const time = Math.round(t1 - t0);
       expect(time).to.be.lessThan(300);
     });
+  });
 
-    it('should render maximum twice', async function test() {
-      let renderCellCount = 0;
-      let renderHeaderCount = 0;
-      const TestCasePerf = () => {
-        const [cols, setCols] = React.useState<GridColDef[]>([]);
-        const data = useData(10, 10);
-
-        React.useEffect(() => {
-          if (data.columns.length) {
-            const newColumns = [...data.columns];
-            newColumns[1].renderHeader = (params) => {
-              renderHeaderCount += 1;
-
-              return <span>{params.field}</span>;
-            };
-            newColumns[1].renderCell = (params) => {
-              if (params.id === 0) {
-                renderCellCount += 1;
-              }
-              return <span>{params.value}</span>;
-            };
-            setCols(newColumns);
-          }
-        }, [data.columns]);
+  it('should prune rendering on cells', () => {
```

Comment:
this test should pass without the perf changes. Have you checked?

Reply:
https://github.com/mui-org/material-ui-x/pull/1662#discussion_r631941114